### PR TITLE
Fix proposal detail page various bugs

### PR DESCRIPTION
**Pull request by ivanng-oursky**
- remaining voting days should be difference in calendar days (if <24hrs remaining, count as one day)
- deposit total needs to be converted from minimal token to token
- don't show zero when value of remaining voting days is zero
- link proposer, voter and depositor to their respective portfolio page

image

### CODE REVIEW

**Comment by ivanng-oursky** [comment]
@hochiw will fix the proposal list change tab wrong page number issue in this PR too.

File: react-app/src/components/ProposalDetailScreen/ProposalDetailScreenAPI.ts [hunk 1]
```diff
@@ -387,11 +392,15 @@ export function useProposalQuery(): {
...
       requestState,
+      // eslint-disable-next-line complexity
       (r) => {
         if (!r.proposalByID) {
           return null;
         }
         const proposal = r.proposalByID;
+        console.log(
+          formatDistance(new Date(proposal.votingEndTime), new Date())
+        );
```
Comment (hochiw) [inline react-app/src/components/ProposalDetailScreen/ProposalDetailScreenAPI.ts:403]: debug log

File: react-app/src/components/ProposalDetailScreen/ProposalDetailScreenAPI.ts [hunk 2]
```diff
@@ -405,10 +414,12 @@ export function useProposalQuery(): {
...
           submitTime: new Date(proposal.submitTime),
           turnout: proposal.turnout ?? null,
-          remainingVotingDays:
+          remainingVotingDuration:
             proposal.votingStartTime &&
             proposal.votingEndTime &&
-            differenceInDays(new Date(proposal.votingEndTime), Date.now()),
+            formatDistance(new Date(proposal.votingEndTime), new Date(), {
+              locale: locale === Locale.zh ? zhTW : undefined,
+            }),
```
Comment (hochiw) [inline react-app/src/components/ProposalDetailScreen/ProposalDetailScreenAPI.ts:422]: Let's create a map in LocaleProvider for this, as we don't know whether there will be more languages in the future

File: react-app/src/components/ProposalDetailScreen/ProposalHeader.tsx [hunk 3]
```diff
@@ -139,6 +142,23 @@ const ProposalTypeAndProposer: React.FC<{ proposal: Proposal }> = ({
   proposal,
 }) => {
   const { type, proposerAddress, submitTime } = proposal;
+
+  const [name, setName] = useState<string>();
```
Comment (hochiw) [inline react-app/src/components/ProposalDetailScreen/ProposalHeader.tsx:146]: vague naming, what name?

**Review by hochiw — CHANGES_REQUESTED** [review]

**Comment by ivanng-oursky** [comment]
@hochiw updated

**Review by hochiw — APPROVED** [review]

**Pull request merged**